c7n/resources/ebs.py

```diff
@@ -79,12 +80,77 @@ class resource_type(object):
     action_registry = ActionRegistry('ebs-snapshot.actions')
 
     def resources(self, query=None):
+        qfilters = QueryParser.parse(self.data.get('query', []))
         query = query or {}
+        if qfilters:
+            query['Filters'] = qfilters
         if query.get('OwnerIds') is None:
             query['OwnerIds'] = ['self']
         return super(Snapshot, self).resources(query=query)
 
 
+class QueryParser(object):
+
+    QuerySchema = {
+        'description': six.string_types,
+        'owner-alias': ('amazon', 'amazon-marketplace', 'microsoft'),
+        'owner-id': six.string_types,
+        'progress': six.string_types,
+        'snapshot-id': six.string_types,
+        'start-time': six.string_types,
+        'status': ('pending', 'completed', 'error'),
+        'tag': six.string_types,
+        'tag-key': six.string_types,
+        'volume-id': six.string_types,
+        'volume-size': six.string_types,
+    }
+
+    @classmethod
+    def parse(cls, data):
```

Comment:
shouldn't this be part of validate?

Reply:
Resource managers don’t participate in validate protocol, they raise on instantiate for error.

Reply:
Resource managers don’t participate in validate protocol, they raise on instantiate for error.